addons/udes_stock_picking_batch/models/stock_picking.py

```diff
@@ -1,5 +1,55 @@
-from odoo import models
+from odoo import models, api
 
 
 class StockPicking(models.Model):
     _inherit = "stock.picking"
+
+    @api.constrains("batch_id")
+    def _trigger_batch_confirm_and_remove(self):
+        """
+       Batch may require new pickings to be auto confirmed or removed
+       Note: this function is very fragile, any change might brake its functionality
+        """
+        orig_batches = self.env.context.get("orig_batches")
+        if orig_batches:
+            for batch in orig_batches:
```

Comment:
Question: Do we need to enter in loop , as from what i see methods inside handles the multi records.

So probably could do direct :  `batches._assign_picks() `etc and just need to find batches first
`batches = orig_batches or self.batch_id `

Reply:
I don't think so but very simple changes in this function were causing unexpected behaviour so I avoided touching it too much. I am going to push a simplified version of it that seems to work now that I have implemented the recompute state on the action_assign on picks.

Reply:
@armandcela This should be addressed now